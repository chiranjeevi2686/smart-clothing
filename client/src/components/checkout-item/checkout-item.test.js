import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react';
import CheckoutItem from './checkout-item.component';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';


const defaultStore = {
    cart: {
        cartItems: [{
            id: 20,
            imageUrl: "https://i.ibb.co/mJS6vz0/blue-jean-jacket.png",
            name: "Blue Jean Jacket",
            price: 90,
            quantity: 1
        }],
        hidden: false,
        payment: null
    }
};

const mockStore = configureStore();
describe('Checkout item component', () => {
    it('should render checkout item', async () => {
        const itemProp = { id: 20, imageUrl: 'www.test.com', price: '23', name: 'test', quantity: '1' }
       // mockStore(defaultStore).dispatch = jest.fn();
       // const action = {payload: {id: 20, imageUrl: 'www.test.com', price: '23', name: 'test', quantity: '1'}, type: 'ADD_ITEM'}
        const { asFragment, getByTestId } = render(<Provider store={mockStore(defaultStore)}>
            <CheckoutItem cartItem={itemProp} />
        </Provider>)
        expect(asFragment()).toMatchSnapshot();
        const addArrow = getByTestId('add-arrow');
        const removeArrow = getByTestId('remove-arrow');
        const removeButton = getByTestId('remove-button');
        expect(getByTestId('add-arrow')).toBeInTheDocument();
        fireEvent.click(addArrow);
        // await waitFor(()=>{
        //     expect(mockStore(defaultStore).dispatch).toHaveBeenCalledWith(action)
        // })
        //expect(mockDispatch).toHaveBeenCalledWith(yourAction);
        fireEvent.click(addArrow);
        fireEvent.click(removeArrow);
        fireEvent.click(removeButton);
        //TODO:need to fix
        //expect(removeButton).not.toBeInTheDocument();
    })
})

