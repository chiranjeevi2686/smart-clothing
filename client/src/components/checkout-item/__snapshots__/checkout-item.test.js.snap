// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkout item component should render checkout item 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ ivfSWR"
  >
    <div
      class="sc-gsTCUz kHrmTY"
    >
      <img
        alt="item"
        src="www.test.com"
      />
    </div>
    <span
      class="sc-dlfnbm cXbNzo"
    >
      test
    </span>
    <span
      class="sc-dlfnbm sc-hKgILt cXbNzo kEVSkE"
    >
      <div
        class="arrow"
        data-test-id="remove-arrow"
      >
        ❮
      </div>
      <span
        class="value"
      >
        1
      </span>
      <div
        class="arrow"
        data-test-id="add-arrow"
      >
        ❯
      </div>
    </span>
    <span
      class="sc-dlfnbm cXbNzo"
    >
      23
    </span>
    <div
      class="sc-eCssSg dJkOIt"
      data-test-id="remove-button"
    >
      ✕
    </div>
  </div>
</DocumentFragment>
`;
